src/java/org/apache/cassandra/schema/CompactionParams.java

```diff
@@ -92,7 +93,7 @@ public static Optional<TombstoneOption> forName(String name)
                                                                                       DEFAULT_ENABLED,
                                                                                       DEFAULT_PROVIDE_OVERLAPPING_TOMBSTONES);
 
-    public static final CompactionParams DEFAULT = CassandraRelevantProperties.UCS_DEFAULT_STRATEGY.getBoolean() ? DEFAULT_UCS.get() : DEFAULT_STCS.get();
+    public static final CompactionParams DEFAULT = DatabaseDescriptor.isEmulateDbaasDefaults() ? DEFAULT_UCS.get() : DEFAULT_STCS.get();
```

Comment:
hmm... this is `static final`, but the flag is perfectly mutable and actually mutated... 

How do we make sure that this `DEFAULT` is not evaluated prematurely?
I suppose that making this a lambda (so that e.g. every table created after setting `isEmulateDbaasDefaults` is guaranteed to use the right compaction strategy) would be safer, but it is also more changes...

I'm sorry, I haven't realized that earlier... I'd feel safer if we had a system property somewhere underneath the flag... WDYT?